src/publisher.js

```diff
@@ -130,21 +155,59 @@ Publisher.prototype.publish = function () {
 
 					http(config, function (error, response) {
 						if (!error && (response.statusCode >= 200 && response.statusCode < 300)) {
-							deferred.resolve();
+							publishDeferred.resolve(data);
 						} else {
 							if (error != null) {
-								deferred.reject(error);
+								publishDeferred.reject(error);
 							} else {
-								deferred.reject(new Error('Unable to publish Pact to Broker: ' + response.statusCode));
+								publishDeferred.reject(new Error('Unable to publish Pact to Broker: ' + response.statusCode));
 							}
 						}
 					});
 
-					return deferred.promise;
+					return publishDeferred.promise;
 				}, function(err) {
 					return q.reject(err);
 				})
+				.then(function(data) {
+					if (!options.tags.length) {
+						deferred.resolve();
+						return deferred.promise;
+					}
+
+					return q.all(_.map(options.tags, function(tag) {
+						var tagDeferred = q.defer();
+						var tagUrl = constructTagUrl(options, tag, data)
 
+						var config = {
+							uri: tagUrl,
+							method: 'PUT',
+							headers: { 'Content-Type': 'application/json' },
+							json: true,
+							auth: auth
+						};
+
+						http(config, function (error, response) {
+							if (!error && response.statusCode >= 200 && response.statusCode < 300) {
+								tagDeferred.resolve();
+							} else {
+								if (error != null) {
+									tagDeferred.reject(error);
+								} else {
+									tagDeferred.reject(new Error('Could not tag Pact with tag "' + tag + '": ' + response.statusCode));
+								}
+							}
+						});
+
+						return tagDeferred.promise;
+					}))
+				}, function(err) {
+					return q.reject(err);
+				})
+				.then(function() {
+					deferred.resolve();
+					return deferred.promise;
```

Comment:
Yeah, you can with Q.denodeify for the http call and then chaining the promises properly.